ui/Url.elm

```diff
@@ -0,0 +1,204 @@
+                    | route = VersionView product channel platform version
+                    , buildIdFilter = ""
+                    , productFilter = Debug.log "product filter" product
+                    , channelFilter = Debug.log "channel filter" channel
+                    , platformFilter = Debug.log "platform filter" platform
+                    , versionFilter = Debug.log "version filter" version
+                    , localeFilter = "all"
+                }
+
+            Just (LocaleView product channel platform version locale) ->
+                { model
+                    | route = VersionView product channel platform version
+                    , buildIdFilter = ""
+                    , productFilter = Debug.log "product filter" product
+                    , channelFilter = Debug.log "channel filter" channel
+                    , platformFilter = Debug.log "platform filter" platform
+                    , versionFilter = Debug.log "version filter" version
+                    , localeFilter = Debug.log "locale fitler" locale
+                }
+
+            _ ->
+                { model
+                    | route = MainView
+                    , buildIdFilter = ""
+                    , productFilter = "all"
+                    , channelFilter = "all"
+                    , platformFilter = "all"
+                    , versionFilter = "all"
+                    , localeFilter = "all"
+                }
+
+
+urlFromRoute : Model -> String
+urlFromRoute model =
+    case model.route of
+        DocsView ->
+            "#/docs/"
+
+        BuildIdView buildId ->
+            "#/builds/buildId/" ++ buildId
+
+        ProductView product ->
+            "#/builds/product/" ++ product
+
+        ChannelView product channel ->
+            "#/builds/product/"
+                ++ product
+                ++ "/channel/"
+                ++ channel
+
+        PlatformView product channel platform ->
+            "#/builds/product/"
+                ++ product
+                ++ "/channel/"
+                ++ channel
+                ++ "/platform/"
+                ++ platform
+
+        VersionView product channel platform version ->
+            "#/builds/product/"
+                ++ product
+                ++ "/channel/"
+                ++ channel
+                ++ "/platform/"
+                ++ platform
+                ++ "/version/"
+                ++ version
+
+        LocaleView product channel platform version locale ->
+            "#/builds/product/"
+                ++ product
+                ++ "/channel/"
+                ++ channel
+                ++ "/platform/"
+                ++ platform
+                ++ "/version/"
+                ++ version
+                ++ "/locale/"
+                ++ locale
+
+        _ ->
+            "#/builds/"
+
+
+routeFromFilters : Model -> Route
+routeFromFilters model =
+    if model.buildIdFilter /= "" then
+        BuildIdView model.buildIdFilter
```

Comment:
Nit: How about destructuring to decrease verbosity here?

```elm
routeFromFilters model =
    let
        { buildIdFilter, productFilter, channelFilter, platformFilter, versionFilter, localeFilter }
            = model
    in
        if buildIdFilter /= "" then
            BuildIdView buildIdFilter
        else if localeFilter /= "all" then
            LocaleView productFilter channelFilter platformFilter versionFilter localeFilter
```

Also it really feels like filters should be a `Dict`, but let's not waste time with this yet.